### ISSUE TITLE
Change URI parameter “deleted” and add new parameter “canceled”

### ISSUE THREAD

**Issue by jirutka** [cvut/sirius#216]
The flag “deleted” in the events table currently has two meanings:

1. Event has been _deleted_ by a change in the source data (i.e. in KOS). This usually happens before start of a semester and it’s basically just a technical information needed for data synchronization. Users don’t need to be informed about such change.
2. Event has been _cancelled_ by a schedule exception. This usually happens during semester and since it’s an _exception_, users should be informed about such change.

The URI parameter “deleted” is quite misleading, because it adds only events deleted by a schedule exception (see dc7d03fbe63da4d020ade5ea093f6aa54fe76ebf). This is problem for clients that do full synchronization and needs to know about all deleted events (e.g. project of M. Kozák from FEL).

A systematic solution would be to add a new parameter “cancelled” for (2) and change “deleted” to include _all_ deleted events (i.e. both cases (1) and (2)). However, this would be breaking change in the API.

Therefore I suggest to:

* Add new URI parameter “cancelled” with the same meaning as the current “deleted” and encourage users to use it instead of “deleted”.
* Change type of the “deleted” parameter from boolean to enum and accept extra value “all” – to include _all_ deleted events.

@jnv, @flexik, WDYT, any better idea?

**Comment by jnv** [comment]
I agree with introducing extra value for `deleted` for maximum backward compatibility and I would focus on that first place.

I would introduce `cancelled` in new API revision with clear break from previous behaviour.

**Comment by jirutka** [comment]
The first part is done, parameter “deleted” now accepts special value “all”.
